cvat/apps/analytics_report/report/create.py

```diff
@@ -0,0 +1,486 @@
+
+    @classmethod
+    def _get_analytics_check_job_delay(cls) -> timedelta:
+        return timedelta(seconds=settings.ANALYTICS_CHECK_JOB_DELAY)
+
+    def _get_scheduler(self):
+        return django_rq.get_scheduler(settings.CVAT_QUEUES.ANALYTICS_REPORTS.value)
+
+    def _get_queue(self):
+        return django_rq.get_queue(settings.CVAT_QUEUES.ANALYTICS_REPORTS.value)
+
+    def _make_queue_job_prefix(self, obj) -> str:
+        if isinstance(obj, Task):
+            return f"{self._QUEUE_JOB_PREFIX_TASK}{obj.id}-"
+        else:
+            return f"{self._QUEUE_JOB_PREFIX_PROJECT}{obj.id}-"
+
+    def _make_custom_analytics_check_job_id(self) -> str:
+        return uuid4().hex
+
+    def _make_initial_queue_job_id(self, obj) -> str:
+        return f"{self._make_queue_job_prefix(obj)}initial"
+
+    def _make_regular_queue_job_id(self, obj, start_time: timezone.datetime) -> str:
+        return f"{self._make_queue_job_prefix(obj)}{start_time.timestamp()}"
+
+    @classmethod
+    def _get_last_report_time(cls, obj):
+        try:
+            report = obj.analytics_report
+            if report:
+                return report.created_date
+        except ObjectDoesNotExist:
+            return None
+
+    def _find_next_job_id(self, existing_job_ids, obj, *, now) -> str:
+        job_id_prefix = self._make_queue_job_prefix(obj)
+
+        def _get_timestamp(job_id: str) -> timezone.datetime:
+            job_timestamp = job_id.split(job_id_prefix, maxsplit=1)[-1]
+            if job_timestamp == "initial":
+                return timezone.datetime.min.replace(tzinfo=timezone.utc)
+            else:
+                return timezone.datetime.fromtimestamp(float(job_timestamp), tz=timezone.utc)
+
+        max_job_id = max(
+            (j for j in existing_job_ids if j.startswith(job_id_prefix)),
+            key=_get_timestamp,
+            default=None,
+        )
+        max_timestamp = _get_timestamp(max_job_id) if max_job_id else None
+
+        last_update_time = self._get_last_report_time(obj)
+        if last_update_time is None:
+            # Report has never been computed, is queued, or is being computed
+            queue_job_id = self._make_initial_queue_job_id(obj)
+        elif max_timestamp is not None and now < max_timestamp:
+            # Reuse the existing next job
+            queue_job_id = max_job_id
+        else:
+            # Add an updating job in the queue in the next time frame
+            delay = self._get_analytics_check_job_delay()
+            intervals = max(1, 1 + (now - last_update_time) // delay)
+            next_update_time = last_update_time + delay * intervals
+            queue_job_id = self._make_regular_queue_job_id(obj, next_update_time)
+
+        return queue_job_id
+
+    class AnalyticsReportsNotAvailable(Exception):
+        pass
+
+    def schedule_analytics_report_autoupdate_job(self, *, job=None, task=None, project=None):
+        if sum(map(bool, (job, task, project))) > 1:
```

Comment:
```suggestion
        assert sum(map(bool, (job, task, project))) == 1
```

This error looks more like an implementation error than a problem-specific one.